packages/devextreme/js/ui/tab_panel.js

```diff
@@ -252,6 +273,51 @@ const TabPanel = MultiView.inherit({
         this._focusTarget().attr('tabIndex', -1);
     },
 
+    _getTabsOrientation() {
+        const { tabsPosition } = this.option();
+
+        if([TABS_POSITION.top, TABS_POSITION.bottom].includes(tabsPosition)) {
+            return TABS_ORIENTATION.horizontal;
+        }
+
+        if([TABS_POSITION.right, TABS_POSITION.left].includes(tabsPosition)) {
+            return TABS_ORIENTATION.vertical;
+        }
+    },
+
+    _getTabPanelTabsPositionClass(value = null) {
+        value = value ?? this.option('tabsPosition');
+
+        switch(value) {
```

Comment:
Why we should have cases for it?
```
const TABPANEL_TABS_POSITION_CLASS = dx-tabpanel-tabs-position;

return `${TABPANEL_TABS_POSITION_CLASS}-${value}`
```

Reply:
There is not such class `dx-tabpanel-tabs-position`. We shouldn't use string templates for strict class variables. May be tomorrow we will use the different variable, so we will need to change js code.